main.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import "fmt"
+
+func worker(id int, jobs <-chan int, results chan<- int) {
+	for j := range jobs {
+		go func() {
+			switch j % 3 {
+			case 0:
+				j = j * 1
+			case 1:
+				j = j * 2
+				results <- j * 2
+			case 2:
+				results <- j * 3
+				j = j * 3
+			}
+		}()
+	}
+}
+func main() {
+	jobs := make(chan int)
+	results := make(chan int)
+	for i := 1; i <= 1000000000; i++ {
+		go func() {
+			if i%2 == 0 {
+				i += 99
+			}
+			jobs <- i
+		}()
+	}
+	close(jobs)
+	jobs2 := []int{}
```

Comment:
Issue: `jobs2` is redundant, since its value is taken to range and call worker job.
Fix: Replace `jobs2` with a const, or directly the value 1000